droplet_autoscaler_test.go

```diff
@@ -0,0 +1,565 @@
+        "max_instances": 5,
+        "target_cpu_utilization": 0.5,
+        "cooldown_minutes": 5
+      },
+      "droplet_template": {
+        "size": "s-1vcpu-512mb-10gb",
+        "region": "s2r1",
+        "image": "547864",
+        "tags": [
+          "test-ag-01"
+        ],
+        "ssh_keys": [
+          "372862",
+          "367582",
+          "355790"
+        ],
+        "vpc_uuid": "72b0812c-7535-4388-8507-5ad29b4487b3",
+        "with_droplet_agent": false,
+        "project_id": "",
+        "ipv6": true,
+        "user_data": "\n#cloud-config\nruncmd:\n- apt-get update\n- apt-get install -y stress-ng\n"
+      },
+      "created_at": "2024-10-21T13:05:23Z",
+      "updated_at": "2024-10-21T13:05:23Z",
+      "current_utilization": {
+        "memory": 0.33,
+        "cpu": 0.0007
+      },
+      "status": "active"
+    },
+    {
+      "id": "1044bfca-e490-44a1-aa1c-6f002daf6a13",
+      "name": "test-autoscalergroup-01",
+      "config": {
+        "min_instances": 1,
+        "max_instances": 5,
+        "target_cpu_utilization": 0.5,
+        "cooldown_minutes": 5
+      },
+      "droplet_template": {
+        "size": "s-1vcpu-512mb-10gb",
+        "region": "s2r1",
+        "image": "547864",
+        "tags": [
+          "test-ag-01"
+        ],
+        "ssh_keys": [
+          "372862",
+          "367582",
+          "355790"
+        ],
+        "vpc_uuid": "72b0812c-7535-4388-8507-5ad29b4487b3",
+        "with_droplet_agent": false,
+        "project_id": "",
+        "ipv6": true,
+        "user_data": "\n#cloud-config\nruncmd:\n- apt-get update\n- apt-get install -y stress-ng\n"
+      },
+      "created_at": "2024-10-18T19:03:08Z",
+      "updated_at": "2024-10-18T19:03:08Z",
+      "current_utilization": {
+        "memory": 0.35,
+        "cpu": 0.0009
+      },
+      "status": "active"
+    },
+    {
+      "id": "b92962b5-26a5-4e63-a1d9-a0f5d44b4f23",
+      "name": "test-autoscalergroup-02",
+      "config": {
+        "min_instances": 1,
+        "max_instances": 5,
+        "target_cpu_utilization": 0.5,
+        "cooldown_minutes": 5
+      },
+      "droplet_template": {
+        "size": "s-1vcpu-512mb-10gb",
+        "region": "s2r1",
+        "image": "547864",
+        "tags": [
+          "test-ag-01"
+        ],
+        "ssh_keys": [
+          "372862",
+          "367582",
+          "355790"
+        ],
+        "vpc_uuid": "72b0812c-7535-4388-8507-5ad29b4487b3",
+        "with_droplet_agent": false,
+        "project_id": "",
+        "ipv6": true,
+        "user_data": "\n#cloud-config\nruncmd:\n- apt-get update\n- apt-get install -y stress-ng\n"
+      },
+      "created_at": "2024-10-21T13:05:12Z",
+      "updated_at": "2024-10-21T13:05:12Z",
+      "current_utilization": {
+        "memory": 0.56,
+        "cpu": 0.0002
+      },
+      "status": "active"
+    }
+  ],
+  "links": {},
+  "meta": {
+    "total": 3
+  }
+}
+`
+
+var dropletAutoscaleGetJSONResponse = `
+{
```

Comment:
Question about the API itself: To remain consistent with the rest of our API, I'd expect the response to have a key like `{"autoscale_pool": { "id": "aaa-bbb-ccc",  ... } }`. Is this already set in stone or do we have an opportunity to change it still?

Reply:
👋 @andrewsomething let me follow up on the API front and confirm, it shouldn't be set in stone and as such we should still be able to tweak this.